packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -462,15 +497,75 @@ export default class OperatorModeStateService extends Service {
         } else {
           dirs.splice(i, 1);
         }
-        this.state.openDirs = dirs;
+        this.openDirs.set(this.realmURL, new TrackedArray(dirs));
         return;
       } else if (entryPath.startsWith(dirs[i])) {
         dirs[i] = entryPath;
-        this.state.openDirs = dirs;
+        this.openDirs.set(this.realmURL, new TrackedArray(dirs));
         return;
       }
     }
-    this.state.openDirs = [...dirs, entryPath];
+    this.openDirs.set(this.realmURL, new TrackedArray([...dirs, entryPath]));
     this.schedulePersist();
   }
+
+  private get readyFile() {
+    if (isReady(this.openFile.current)) {
+      return this.openFile.current;
+    }
+    throw new Error(
+      `cannot access file contents ${this.state.codePath} before file is open`,
+    );
+  }
+
+  get openFileIsReady() {
+    return isReady(this.openFile.current);
+  }
+
+  get realmURL() {
```

Comment:
Here, 

`this.realmURL` is a string
`this.cachedRealmURL` is a URL

Should we make this more consistent in terms of the return type? I think the preferred type is to operate with `URL` with getters and params

Reply:
You’re right, I was dreading it but it turned out okay with the help of Typescript, fixed now, thanks!